python/graphstorm/gconstruct/remap_result.py

```diff
@@ -0,0 +1,477 @@
+                        args.preserve_input)
+
+
+def add_distributed_remap_args(parser):
+    """ Distributed remapping only
+
+        Users can ignore arguments in this argument group.
+        The arguments under this argument group are mainly
+        designed for distributed remapping results in SageMaker,
+        where a shared file system is not avaliable.
+
+        Parameter
+        ---------
+        parser: argparse.ArgumentParser
+            Argument parser
+
+        Return
+        ------
+        parser: Argument parser
+    """
+    group = parser.add_argument_group(title="dist_remap")
+    group.add_argument("--with-shared-fs",
+                       type=lambda x: (str(x).lower() in ['true', '1']),default=True,
+                       help="Whether all files are stored in a shared file system"
+                            "False when it is running on SageMaker")
+
+    group.add_argument("--rank", type=int, default=0,
+                           help="The rank of current worker.")
+    group.add_argument("--world-size", type=int, default=1,
+                           help="Totoal number of workers in the cluster.")
+
+    return parser
+
+def generate_parser():
+    """ Generate an argument parser
+    """
+    parser = argparse.ArgumentParser("Remapping graph node IDs")
+    parser.add_argument(
+        "--yaml_config_file",
+        "--cf",
+        help="pointer to the yaml configuration file of the experiment",
+        type=str,
+        default=None
+    )
+
+    group = parser.add_argument_group(title="remap")
+    group.add_argument("--num-processes", type=int, default=4,
+                       help="The number of processes to process the data simultaneously.")
+    group.add_argument("--node-id-mapping", type=str,
+                       help="The directory storing the id mappings")
+    group.add_argument("--prediction-dir", type=str,
+                       help="The directory storing the graph prediction results.")
+    group.add_argument("--output-format", type=str,
+                       default="parquet",
+                       choices=['parquet'],
+                       help="The format of the output.")
+    group.add_argument("--logging-level", type=str, default="info",
+                       help="The logging level. The possible values: debug, info, warning, \
+                                   error. The default value is info.")
+
+    group.add_argument("--pred-etypes", type=str, nargs="+", default=None,
+                       help="[Optional] A list of canonical edge types which have"
+                                "prediction results For example, "
+                                "--pred-etypes user,rate,movie user,watch,movie"
+                                "If pred_etypes is not provided, result_info.json "
+                                "under prediction_dir will be used to retrive the pred_etypes")
+    group.add_argument("--output-chunk-size", type=int, default=100000,
```

Comment:
why do we want to output the remapped results in a different number of files?

Reply:
Currently the number of prediction result or embedding files are same as the number of trainers. 
When the graph is large, it is possible that a single output file can be tens of GBs or even larger.
With this parameter, we can split those files into smaller files. For example thousands of parquet files, each of which takes only hundreds of MB space.

Reply:
i see. the default behavior should be not splitting the files.